composer/callbacks/checkpoint_saver.py

```diff
@@ -274,7 +274,13 @@ def __init__(
 
         is_remote_folder = backend != ''
         if is_remote_folder:  # If uploading to a remote path, use a temporary directory to save local checkpoints.
-            local_folder = os.path.join(tempfile.mkdtemp(), local_folder)
+            root_temp_folder = None
```

Comment:
I believe this is already how mkdtemp works? `https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp`. The first place it should look to decide the tmpdir folder is whatever is specified by the `TMPDIR` env var.